scripts/jenkins/ardana/gerrit/gerrit.py

```diff
@@ -0,0 +1,235 @@
+import json
+import re
+
+try:
+    from pygerrit2 import GerritRestAPI, GerritReview, HTTPDigestAuthFromNetrc
+except ImportError:
+    # Only reviewing and merging functionality depends on pygerrit2
+    pass
+
+import requests
+
+GERRIT_URL = 'https://gerrit.suse.provo.cloud'
+GERRIT_VERIFY = True
+
+# We use a more complex regex that matches both formats of Depends-On so that
+# we preserve the order in which they are discovered.
+DEPENDS_ON_RE = re.compile(
+    r"^\W*Depends-On: ("
+    r"((http(s)?:\/\/)?gerrit\.suse\.provo\.cloud\/(\/)?(\#\/c\/)?(\d+).*?)"
+    r"|(I[0-9a-f]{40})"
+    r")\s*$",
+    re.MULTILINE | re.IGNORECASE)
+
+
+def find_dependency_headers(message):
```

Comment:
We should move this into GerritChange as a private method to avoid it being used elsewhere and having dependency objects recreated (see above comments).

Reply:
Hmm, thinking about it, I wonder if GerritChange::get_dependencies should actually walk the commit history to find any other dependencies? This will ensure they are loaded, along with their own dependencies, and then their packages are built if necessary.

It also looks like OBSPackage::get_depends_on_change isn't called from anywhere any more unless I've missed something. So it's likely there is a regression there.

Reply:
`get_depends_on_change` isn't called from anywhere anymore, because I have mistakenly replaced it with `GerritChange::get_dependencies`. I'll fix that right away, but as far as making `GerritChange::get_dependencies` to walk the commit history, that would be really painful to achieve without a local git clone and I don't see that functionality having value anywhere but during the test package building, while the current implementation of `get_dependencies` is still used in at least one other spot.

Reply:
Won't it still be necessary for changes that haven't merged? A change can come into gerrit with related changes that are commits between the target branch and itself. For example, when somebody pushes multiple commits into gerrit, they don't specify "depends-on" for each parent commit. These patches are all pulled in when the change is checked out, so we get them all. However, if one of the middle commits has a cross-repo depends on, then that will have been missed.